catroid/src/main/java/org/catrobat/catroid/content/eventids/EventId.java

```diff
@@ -22,7 +22,48 @@
  */
 package org.catrobat.catroid.content.eventids;
 
-public abstract class EventId {
-	public abstract boolean equals(Object o);
-	public abstract int hashCode();
+import android.support.annotation.IntDef;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+public class EventId {
+	@Retention(RetentionPolicy.SOURCE)
+	@IntDef({TAP, TAP_BACKGROUND, OTHER})
+	@interface EventType {
+	}
+
+	public static final int OTHER = 0;
+	public static final int TAP = 1;
+	public static final int TAP_BACKGROUND = 2;
+
+	@EventType
+	private final int type;
+
+	public EventId(int type) {
```

Comment:
if theres only EventType ints allowed for this, please add @EventType to the parameter.

public EventId(@EventType int type)